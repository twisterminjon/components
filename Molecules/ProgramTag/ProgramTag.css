.programtag {
  position: relative;
  display: inline-block;

  border: none;
  border-radius: 6px;

  background-color: var(--brandcolor_dark);
  color: var(--white);

  font-size: 14px;
  line-height: 30px;
}

.programtag-checkboxcontainer {
  display: flex;
  align-items: center;
  height: 100%;
  margin-left: 8px;
  width: 15px;
  height: 15px;
}

.programtag-textcontainer {
  margin-left: 8px;
  width: 100%;
}

.programtag--fluid {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.programtag-button {
  height: 30px;
  width: 30px;
}

.programtag--loading {
  color: var(--button_color_disabled);
  pointer-events: none;
}

.programtag--inactive {
  color: var(--button_color_disabled);
}
