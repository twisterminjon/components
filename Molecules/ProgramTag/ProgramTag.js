import React, { Component } from 'react'
import PropTypes from 'prop-types'
import EditProgramButton from '../../Atoms/EditProgramButton/EditProgramButton'
import IconSquareOpenSmall from '../../Atoms/Icons/IconSquareOpenSmall'
import IconCheckSmall from '../../Atoms/Icons/IconCheckSmall'

import './ProgramTag.css'

const renderCheckboxIcon = (value, disabled) => {
  const iconStyle = {
    display: 'block',
  }
  return (
    <React.Fragment>
      {!value && (
        <IconSquareOpenSmall
          style={iconStyle}
          color={disabled ? 'var(--checkbox__icon_disabled_fg)' : 'var(--checkbox__icon_enabled_fg)'}
        />
      )}
      {value && (
        <IconCheckSmall
          style={iconStyle}
          color={disabled ? 'var(--checkbox__icon_disabled_fg)' : 'var(--checkbox__icon_enabled_fg)'}
        />
      )}
    </React.Fragment>
  )
}

export default class ProgramTag extends Component {
  static propTypes = {
    /** Text to display in the tag */
    label: PropTypes.string.isRequired,
    /** id of the item being displayed */
    id: PropTypes.string.isRequired,
    /** The id of the enrollment */
    enrollmentId: PropTypes.string.isRequired,
    /** A tag can display an inactive state */
    inActive: PropTypes.bool,
    /** Called when the edit button is clicked */
    onEdit: PropTypes.func.isRequired,
    /** ClassName for the wrapper */
    className: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
    /** Can show a loading spinner */
    loading: PropTypes.bool,
    /** Represents if program is completed */
    isComplete: PropTypes.bool.isRequired,
  }
  static defaultProps = {
    loading: false,
    className: '',
    style: {},
  }

  state = {
    inActive: false,
  }

  handleEdit = () => {
    this.props.onEdit(this.props.enrollmentId, this.props.id, this.props.label)
  }

  render() {
    const { label, inActive, className, style, loading, isComplete } = this.props

    const loadingClass = loading ? 'programtag--loading button-spinner' : ''
    const inActiveClass = inActive ? 'programtag--inactive' : ''

    return (
      <div
        className={`programtag programtag--fluid ${inActiveClass} ${loadingClass} ${className}`}
        style={style}
        data-testid={`programtag-${label}`}>
        <div className="programtag-checkboxcontainer">{renderCheckboxIcon(isComplete, inActive)}</div>
        <div className="programtag-textcontainer">
          <span data-testid={`programtag-${label}-text`}>{label}</span>
        </div>
        <EditProgramButton
          onClick={() => this.handleEdit()}
          className="programtag-button"
          data-testid={`programtag-${label}-edit-button`}
          disabled={loading}
        />
      </div>
    )
  }
}
