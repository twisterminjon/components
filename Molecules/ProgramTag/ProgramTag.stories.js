import React from 'react'

import { storiesOf } from '@storybook/react'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'
import centered from '@storybook/addon-centered/react'

import ProgramTag from './ProgramTag'

const stories = storiesOf('Provider/v1/Molecules', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('ProgramTag', () => {
  const label = text('label', 'D-list super heros')
  const inActive = boolean('inActive', false)
  const loading = boolean('loading', false)
  const isComplete = boolean('isComplete', false)

  return (
    <ProgramTag
      label={label}
      id="1"
      enrollmentId="11"
      inActive={inActive}
      onEdit={(enrollmentId, id, name) => {
        alert(`edit tag enrollmentId: ${enrollmentId}, id: ${id}, name: ${name}`)
      }}
      loading={loading}
      isComplete={isComplete}
    />
  )
})
