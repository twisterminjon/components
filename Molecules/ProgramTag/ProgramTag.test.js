import React from 'react'
import ProgramTag from './ProgramTag'

describe('ProgramTag', () => {
  const label = 'Legit'
  const id = '1'
  const enrollmentId = '234'
  const mockFun = jest.fn()
  const isComplete = false

  it('matches the snapshot', () => {
    const wrapper = window.shallow(
      <ProgramTag
        id={id}
        label={label}
        enrollmentId={enrollmentId}
        inActive={false}
        isComplete={isComplete}
        onEdit={mockFun}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot when inActive', () => {
    const wrapper = window.shallow(
      <ProgramTag
        id={id}
        label={label}
        enrollmentId={enrollmentId}
        inActive={true}
        isComplete={isComplete}
        onEdit={mockFun}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('displays the correct text', () => {
    const wrapper = window.shallow(
      <ProgramTag
        id={id}
        label={label}
        enrollmentId={enrollmentId}
        inActive={false}
        isComplete={isComplete}
        onEdit={mockFun}
      />
    )

    expect(wrapper.find('[data-testid="programtag-Legit-text"]').text()).toEqual(label)
  })

  it('calls the edit function', () => {
    const wrapper = window.shallow(
      <ProgramTag
        id={id}
        label={label}
        enrollmentId={enrollmentId}
        inActive={false}
        isComplete={isComplete}
        onEdit={mockFun}
      />
    )

    wrapper.find('[data-testid="programtag-Legit-edit-button"]').simulate('click')
    expect(mockFun).toHaveBeenCalled()
  })
})
