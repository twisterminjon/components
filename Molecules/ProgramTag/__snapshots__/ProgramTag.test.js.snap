// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramTag matches the snapshot 1`] = `
<div
  className="programtag programtag--fluid   "
  data-testid="programtag-Legit"
  style={Object {}}
>
  <div
    className="programtag-checkboxcontainer"
  >
    <IconSquareOpenSmall
      color="var(--checkbox__icon_enabled_fg)"
      style={
        Object {
          "display": "block",
        }
      }
    />
  </div>
  <div
    className="programtag-textcontainer"
  >
    <span
      data-testid="programtag-Legit-text"
    >
      Legit
    </span>
  </div>
  <EditProgramButton
    className="programtag-button"
    data-testid="programtag-Legit-edit-button"
    disabled={false}
    onClick={[Function]}
  />
</div>
`;

exports[`ProgramTag matches the snapshot when inActive 1`] = `
<div
  className="programtag programtag--fluid programtag--inactive  "
  data-testid="programtag-Legit"
  style={Object {}}
>
  <div
    className="programtag-checkboxcontainer"
  >
    <IconSquareOpenSmall
      color="var(--checkbox__icon_disabled_fg)"
      style={
        Object {
          "display": "block",
        }
      }
    />
  </div>
  <div
    className="programtag-textcontainer"
  >
    <span
      data-testid="programtag-Legit-text"
    >
      Legit
    </span>
  </div>
  <EditProgramButton
    className="programtag-button"
    data-testid="programtag-Legit-edit-button"
    disabled={false}
    onClick={[Function]}
  />
</div>
`;
